src/lib/distances.jl

```diff
@@ -59,26 +59,30 @@ end
 @adjoint function colwise(s::Euclidean, x::AbstractMatrix, y::AbstractMatrix)
   d = colwise(s, x, y)
   return d, function (Δ::AbstractVector)
-    x̄ = (Δ ./ d)' .* (x .- y)
+    x̄ = (Δ ./ max.(d, eps(eltype(d))))' .* (x .- y)
```

Comment:
I wonder if we can do a similar trick here

Reply:
Something like this would work:
```julia
(Δ .* map(di->ifelse(iszero(di), di, inv(di)), d))' .* (x .- y)
```